pkg/validate/scoped/hydrate/namespace_hydrator.go

```diff
@@ -112,7 +112,6 @@ func buildSelectorMap(ctx context.Context, c client.Client, objs *objects.Scoped
 				hasDynamicNSSelector = true
 			} else if nsSelector.Spec.Mode != "" && nsSelector.Spec.Mode != v1.NSSelectorStaticMode {
```

Comment:
It does seem right that we shouldn't set `hasDynamicNSSelector = true` here, but I don't think this is why the test is flaking.

After taking a closer look, I see that the test is using a shared variable with other test cases in the file. Since these tests run in parallel, it's not safe to use a shared variable in this manner. Each test should get its own copy of the object.

https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testcases/namespace_selectors_test.go#L48